content/articles/build-vue-application-with-django-rest-api-and-axios/index.md

```diff
@@ -0,0 +1,344 @@
+---
+layout: engineering-education
+status: draft
+published: false
+url: /build-vue-application-with-django-rest-api-and-axios/
+title: Building Vue Application with Django REST API and Axios
+description: The objective of this tutorial is to help the reader understand how we can build vue web applications that can consume and display data from API using the Axios library.
+author: atonya-dennis
+date: 2021-12-04T00:00:00-06:40
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/build-vue-application-with-django-rest-api-and-axios/hero.jpg
+    alt: Vue example image
+---
+When developing applications, it's common to use APIs to consume and display data. We can make this possible using the Axios library, that can be used to create Vue.js apps where Django, a python-based backend framework, is only used as an API using the Django rest framework package.
+<!--more-->
+To help you grasp this, we'll cover how to use Vue, Axios, and the Django REST APIs to consume and display data from APIs.
+
+### Table of contents
+- [Table of contents](#table-of-contents)
+- [Prerequisites](#prerequisites)
+- [Introduction](#introduction)
+- [Setting Up Django and Vue](#setting-up-django-and-vue)
+- [Creating the Django App](#creating-the-django-app)
+  - [Django TodoApp Project](#django-todoapp-project)
+  - [Django TodoApp Models](#django-todoapp-models)
+  - [API Accessibility by Vue App](#api-accessibility-by-vue-app)
+- [Creating the Vue App](#creating-the-vue-app)
+- [Getting and Displaying App Data](#getting-and-displaying-app-data)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+The reader should do the following to follow along with this tutorial:
+-Have a fundamental understanding of the Django REST framework and REST API.
+- Have knowledge about Vue.js and the Axios library.
+- Know how to use PyCharm,Visual Studio Code or any other IDE for working with Python Django and Vue js.
+
+Go to this [website](https://www.jetbrains.com/pycharm/download/#section=windows) to download and install the PyCharm IDE on your computer.
+
+
+### Introduction
+**Django** is a Python-based backend framework that may be used to create  applications by displaying HTML code or as a pure API using the Django rest framework package. We can also combine it with Vue.js, a JavaScript-based front-end framework, in building the best sites and applications.
+This is made possible by including the Vue script directly into the Django template HTML code, but we can also use the Django API to consume and display data using the HTTP client library called **Axios**.
+To implement this, we'll create a simple Vue Todo Application together with the Django REST API and Axios.
+
+### Setting Up Django and Vue
+We'll start by setting up the Django and Vue.js in our machines together with the Axios library.
```

Comment:
Minimize the suer of contractions in formal writing. Do this across the entire artile.